routes/problems_id.js

```diff
@@ -58,12 +58,16 @@ module.exports = function(app) {
   app.post('/problems/:id', async function(request, response) {
     const user  = await auth.getUser(request.body.authToken);
     if (user == null) {
+      response.status(400);
```

Comment:
nit: 401 or 403 might better express that the user is not authenticated, also this can be done in one line like this `res.status(401).send('Not Authenticated')`

Reply:
Thanks for noting this!
I'll change it according to your suggestion.